### DISCUSSION TITLE
How to mock wrap().assign() functions for testing an update method.

### DISCUSSION THREAD

**Discussion by elsanto079** [mikro-orm/mikro-orm#3909] [Q&A]
I want to test my service and it generates an error when I want to call the wrap() function.

in yours repo examples the update test is missing: [example]( https://github.com/mikro-orm/nestjs-realworld-example-app/blob/master/src/user/test/user.service.spec.ts)

My service method:

```
async edit(editEventEntity: EditEventEntity): Promise<EventEntity> { 
let event = await this.em.findOne(EventDto, editEventEntity.id);
wrap(event).assign({ ...editEventEntity, updatedBy: user.id });
await this.em.flush();
return event;                  
    }
```

my test file:
```
describe('Happy Paths', () => {
        const mockEntityManager = {
            findOne: jest
                .fn()
                .mockImplementation(
                    (EventDto, id: string): EventDto => {
                        return eventsMock.find(
                            (event) => event.id === id
                        );
                    }),           
            wrap: jest.fn().mockImplementation((): Promise<void> => {
                return;
            }),
            assign: jest.fn().mockImplementation((): Promise<void> => {
                return;
            }),
            flush: jest.fn().mockImplementation((): Promise<void> => {
                return;
            })
        };
        beforeEach(async () => {
            const module: TestingModule = await createTestingModule(
                mockEntityManager
            );
            eventsService = module.get<EventsService>(EventsService);
        });
        describe('Edit event', () => {
            it('should edit an event', async () => {
                const eventEntity: EditEventEntity = {
                    ...editEventMock,
                    externalUserId: mockUserDecoded.externalUserId
                };

                await eventsService.edit(eventEntity);
               
                expect(mockEntityManager.findOne).toHaveBeenCalled();
                expect(mockEntityManager.flush).toHaveBeenCalled();
            });
        });
    });
```
error:  **`HttpException: (0 , core_1.wrap)(...).assign is not a function`**

**Comment by B4nan** [comment]
The right question is, why would you want to mock the assign method? Even better one - what do you actually want to test?

I am not really a big fan of mocking anything in the ORM, you will only make your tests super complex and end up testing your own test implementation, not the actual app.

**Reply by elsanto079** [reply]
Hi, thanks for the quick response, what I want to achieve is test my service functions and mock MikroOrm entitymanager. In this case, my service funtion is in charge of updating the data of an entity and saving the changes in the database.

**Reply by elsanto079** [reply]
I saw on an [issue ](https://github.com/mikro-orm/mikro-orm/issues/1406) that you commented another way of using assign() 
so i replace **wrap().assign()** with **em.assign()** and it was easy to mock 
```
const book = await em.findOneOrFail(Book, 1);
 //wrap(book).assign(new_version, { mergeObjects: true }); 
 em.assign(book, new_version, { mergeObjects: true })
 await em.flush();
```

**Reply by B4nan** [reply]
I really don't think you should be mocking the `assign` helper, it does not make any sense. It mutates the entity object, it is a sync method and it does not touch the database anyhow.
